src/decode.rs

```diff
@@ -1963,75 +1963,38 @@ unsafe extern "C" fn dav1d_msac_decode_uniform(
             .wrapping_add(dav1d_msac_decode_bool_equi(s))
     }) as libc::c_int;
 }
-unsafe extern "C" fn init_quant_tables(
-    seq_hdr: *const Dav1dSequenceHeader,
-    frame_hdr: *const Dav1dFrameHeader,
+
+fn init_quant_tables(
+    seq_hdr: &Dav1dSequenceHeader,
+    frame_hdr: &Dav1dFrameHeader,
     qidx: libc::c_int,
-    mut dq: *mut [[uint16_t; 2]; 3],
+    mut dq: &mut [[[uint16_t; 2]; 3]],
 ) {
-    let mut i: libc::c_int = 0 as libc::c_int;
-    while i
-        < (if (*frame_hdr).segmentation.enabled != 0 {
-            8 as libc::c_int
-        } else {
-            1 as libc::c_int
-        })
-    {
-        let yac: libc::c_int = if (*frame_hdr).segmentation.enabled != 0 {
-            iclip_u8(qidx + (*frame_hdr).segmentation.seg_data.d[i as usize].delta_q)
+    // Safety: `dav1d_dq_tbl: [[[u16; 2]; 256]; 3]` is valid for all bit patterns,
+    // as ultimately they're all `u16`s.
+    let tbl = unsafe { &dav1d_dq_tbl };
+
+    let segmentation_is_enabled = frame_hdr.segmentation.enabled != 0;
+    for i in 0..(if segmentation_is_enabled { 8 } else { 1 }) {
```

Comment:
Fixed in 3dfacb4eed9384b58418246abb267d04fb5bf69f, but I named it `len` instead of `range`; I would name `0..len` `range`, for example, not just one end.